File: Week 1 C/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{

    long number = get_long("Number: ");

    char str[20];

    sprintf(str, "%li", number);

    long length = strlen(str);

    char first_digit = str[0];
    char second_digit = str[1];

    if ((first_digit == '3' && second_digit == '4' && length == 15) || (first_digit == '3' && second_digit == '7' && length == 15))
    {
        printf("AMEX\n");
    }
    else if ((first_digit == '5' && second_digit == '1' && length == 16) ||
             (first_digit == '5' && second_digit == '2' && length == 16) ||
             (first_digit == '5' && second_digit == '3' && length == 16) ||
             (first_digit == '5' && second_digit == '4' && length == 16) ||
             (first_digit == '5' && second_digit == '5' && length == 16))
    {
        printf("MASTERCARD\n");
    }
    else if ((first_digit == '4') && (length == 13 || length == 16))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Week 6 Python/credit/credit.py
import re
from cs50 import get_string

number = get_string("Number: ")

if re.match(r"^(3[47]\d{13})$", number):
    print("AMEX")
elif re.match(r"^(5[1-5]\d{14})$", number):
    print("MASTERCARD")
elif re.match(r"^(4(\d{12}|\d{15}))$", number):
    print("VISA")
else:
    print("INVALID")
